x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -0,0 +1,298 @@
+import org.elasticsearch.action.admin.indices.create.CreateIndexResponse;
+import org.elasticsearch.action.admin.indices.get.GetIndexRequest;
+import org.elasticsearch.action.admin.indices.get.GetIndexResponse;
+import org.elasticsearch.action.admin.indices.refresh.RefreshRequest;
+import org.elasticsearch.action.admin.indices.refresh.RefreshResponse;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.client.Client;
+import org.elasticsearch.cluster.metadata.AliasMetaData;
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.metadata.MappingMetaData;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.collect.ImmutableOpenMap;
+import org.elasticsearch.common.collect.MapBuilder;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.reindex.BulkByScrollResponse;
+import org.elasticsearch.index.reindex.ReindexAction;
+import org.elasticsearch.index.reindex.ReindexRequest;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+
+public class EnrichPolicyRunner {
+
+    private static final Logger logger = LogManager.getLogger(EnrichPolicyRunner.class);
+
+    private ClusterService clusterService;
+    private Client client;
+    private EnrichStore enrichStore;
+    private IndexNameExpressionResolver indexNameExpressionResolver;
+    private LongSupplier nowSupplier;
+
+    EnrichPolicyRunner(ClusterService clusterService, Client client, EnrichStore enrichStore,
+                              IndexNameExpressionResolver indexNameExpressionResolver,
+                              LongSupplier nowSupplier) {
+        this.clusterService = clusterService;
+        this.client = client;
+        this.enrichStore = enrichStore;
+        this.indexNameExpressionResolver = indexNameExpressionResolver;
+        this.nowSupplier = nowSupplier;
+    }
+
+    public void runPolicy(final String policyId, final ActionListener<EnrichPolicyResult> listener) {
+        // Look up policy in policy store and execute it
+        runPolicy(policyId, enrichStore.getPolicy(policyId), listener);
+    }
+
+    public void runPolicy(final String policyName, final EnrichPolicy policy,
+                          final ActionListener<EnrichPolicyResult> listener) {
+        // TODO: Index Pattern expansion and Alias support
+        // Collect the source index information
+        logger.info("Policy [{}]: Running enrich policy", policyName);
+        final String sourceIndexPattern = policy.getIndexPattern();
+        logger.debug("Policy [{}]: Checking source index [{}]", policyName, sourceIndexPattern);
+        GetIndexRequest getIndexRequest = new GetIndexRequest().indices(sourceIndexPattern);
+        client.admin().indices().getIndex(getIndexRequest, new ActionListener<GetIndexResponse>() {
```

Comment:
new ActionListener<GetIndexResponse> -> new ActionListener<>

Reply:
To be clear here, do you mean `new ActionListener<GetIndexResponse>()` -> `new ActionListener<>()`? That change would end up with compilation errors

Reply:
this: `client.admin().indices().getIndex(getIndexRequest, new ActionListener<>() {` does compile here without an error.

Reply:
I get `Cannot use "<>" with anonymous inner classes` in IDEA, and the `onResponse` override only accepts `Object` as its argument. Maybe I'm doing something wrong. Looking at some random `ActionListener` anonymous implementations and they all declare a type instead of using the diamond operator.

Reply:
Strange, IDEA seems to be the one with the hangup. Gradle seems to build this just fine.

Reply:
Odd. It compiled for me in both IDEA and gradle. Update IDEA?